src/main/webapp/components/sources-info/sources-info.js

```diff
@@ -44,41 +55,77 @@ const sourcesMessage = offlineCount => {
 
   return `${offlineCount} sources are currently down`
 }
-
 export const SourcesInfo = props => {
   const sources = props.sources === undefined ? [] : props.sources
-  const offlineCount = sources.filter(source => !source.isAvailable).length
-  const getIcon = source => (source.isAvailable ? OnlineIcon : OfflineIcon)
-  if (props.loading) {
+  const SourceSet = new Set(sources);
+  const getIcon = source => {
+    if(source.__typename == 'FacetResult'){
```

Comment:
We probably shouldn't be using `__typename` inside our react components because it bleeds GraphQL implementation details

Reply:
A better alternative might be to append an extra property to sources from the container and based off where the source came from. I would suggest potentially using an `Icon` property where you specify which icon you want to represent that source. That way, each program can define their own `getIcon` logic from within the container